src/main/java/Duke.java

```diff
@@ -37,342 +48,367 @@ public void run() throws DukeException {
-                            inputs.taskFind(nextLine);
-                        } else if (nextLine.equals("list")) {
-                            if (inputs.inputs.size() == 0) {
-                                System.out.println("No tasks in list");
+            String keyword = nextLine.substring(5);
+            int len = inputs.size();
+            for(int i = 0; i < len; i++) {
+                Input input = inputs.get(i);
+                if (input.content.contains(keyword)) {
+                    if (input.done) {
+                        System.out.println((i+1) + ". " + input.id + "[/] " + input.content +
+                                input.printTime);
+                    } else {
+                        System.out.println((i+1) + ". " + input.id + "[x] " + input.content +
+                                input.printTime);
+                    }
+                }
+            }
+        }
+    }
+
+    // UI class deals with User Input
+    public static class UI {
+        Scanner sc = new Scanner(System.in);
+
+        // UI Constructor, prints DUKE as part of initialization
+        UI() {
+            String logo = " ____        _        \n"
+                    + "|  _ \\ _   _| | _____ \n"
+                    + "| | | | | | | |/ / _ \\\n"
+                    + "| |_| | |_| |   <  __/\n"
+                    + "|____/ \\__,_|_|\\_\\___|\n";
+            System.out.println("Hello from\n" + logo);
+
+            System.out.println("Hello! I'm Duke \n"
+                    + "What can I do for you?");
+        }
+
+        // Reads user input using scanner
+        String readInput() {
+            return this.sc.nextLine();
+        }
+
+        // Prints error if DukeException is caught when constructing Duke Object
+        void showLoadingError() {
+            System.out.println("Loading Error!");
+        }
+
+
+    }
+
+    //Parser class deals with making sense of user command
+    public static class Parser {
+        String nextLine;
+        TaskList inputs;
+
+        Parser(String nextLine, TaskList inputs) {
+            this.nextLine = nextLine;
+            this.inputs = inputs;
+        }
+
+        // Guides the TaskList based on user command
+        void parse() throws DukeException {
+            try {
+                if (nextLine.startsWith("done")) {
+                    inputs.taskDone(nextLine);
+                } else if (nextLine.startsWith("remove")) {
+                    inputs.taskRemove(nextLine);
+                } else if (nextLine.startsWith("todo")) {
+                    inputs.taskTodo(nextLine);
+                } else if (nextLine.startsWith("deadline")) {
+                    inputs.taskDeadline(nextLine);
+                } else if (nextLine.startsWith("event")) {
+                    inputs.taskEvent(nextLine);
+                } else if (nextLine.startsWith("find")) {
+                    inputs.taskFind(nextLine);
+                } else if (nextLine.equals("list")) {
```

Comment:
You could try making these code more succint by nesting lesser conditionals together!